### PR TITLE
Chore: Port oauth token service to identity requester

### PR DESCRIPTION
**Pull request by Jguer**
Chore: Port oauth token service to identity requester

### CODE REVIEW

File: pkg/services/oauthtoken/oauth_token.go [hunk 1]
```diff
@@ -52,20 +53,32 @@ func ProvideService(socialService social.Service, authInfoService login.AuthInfo
...
-	if usr == nil {
+func (o *Service) GetCurrentOAuthToken(ctx context.Context, usr identity.Requester) *oauth2.Token {
+	if usr == nil || usr.IsNil() {
 		// No user, therefore no token
 		return nil
 	}
 
-	authInfoQuery := &login.GetAuthInfoQuery{UserId: usr.UserID}
+	namespace, id := usr.GetNamespacedID()
+	if namespace != identity.NamespaceUser && namespace != identity.NamespaceRenderService {
```
Comment (kalleep) [inline pkg/services/oauthtoken/oauth_token.go:63]: Is this always safe for render service namespace?

File: pkg/services/oauthtoken/oauth_token_test.go [hunk 2]
```diff
@@ -72,6 +72,7 @@ func TestService_HasOAuthEntry(t *testing.T) {
 		},
 	}
 	for _, tc := range testCases {
+		tc := tc
```
Comment (linoman) [inline pkg/services/oauthtoken/oauth_token_test.go:75]: Is this needed?

**Review by linoman — APPROVED** [review]
LGTM. I left one question, it is up to you to address it. Thank you

File: pkg/services/oauthtoken/oauth_token.go (hunk 1 above)
Comment (Jguer) [reply pkg/services/oauthtoken/oauth_token.go:63]: render namespace can be user ID 0, I'll change it

File: pkg/services/oauthtoken/oauth_token_test.go (hunk 2 above)
Comment (Jguer) [reply pkg/services/oauthtoken/oauth_token_test.go:75]: to silence memory aliasing warning from linters :)

**Review by kalleep — APPROVED** [review]

**Pull request merged**